### PR TITLE
sort: stop triggering for single input query on sort_goodie

### PR DESCRIPTION
**Pull request by NaveenKarippai**
###### What does your Pull Request do (check all that apply)?

Choose the most relevant items and use the following title template to name
your Pull Request.

- [ ] New Instant Answer
    - [ ] Cheat Sheets: **`New {Cheat Sheet Name} Cheat Sheet`**
    - [ ] Other: **`New {IA Name} Instant Answer`**
- [x] Improvement
    - [ ] Bug fix: **`{IA Name}: Fix {Issue number or one-line description}`**
    - [x] Enhancement: **`trigger for query input > 1`**
- [ ] Non–Instant Answer
    - [ ] Other (Role, Template, Test, Documentation, etc.): **`{GoodieRole/Templates/Tests/Docs}: {Short Description}`**

###### Description of changes

The sort_goodie should not be triggered for single numbers (input)
query. This could be fixed by using a return statement check on handler.

Provide an overview of the changes this pull request introduces.

* goodie not triggered for single number (input)

###### Which issues (if any) does this fix?

See also: #3144

Fixes #NNNN - how specifically does it fix the issue?

###### People to notify (@GuiltyDolphin)

---

Instant Answer Page: https://duck.co/ia/view/sort

[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @koosha--

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: lib/DDG/Goodie/Sort.pm [hunk 1]
```diff
@@ -38,7 +38,8 @@ handle remainder => sub {
 
     my $count = 0;
     @numbers = map { 0 + $_ } grep { ++$count <= MAX_LIST_SIZE } @numbers;    # Normalize and limit list size.
-
+    return unless $count > 1;
```
Comment (GuiltyDolphin) [inline lib/DDG/Goodie/Sort.pm]: Why not `return unless @numbers > 1`?

Comment (NaveenKarippai) [inline lib/DDG/Goodie/Sort.pm]: @GuiltyDolphin true! Thanks for the review and please check the new changeset.

File: lib/DDG/Goodie/Sort.pm [hunk 2]
```diff
@@ -34,11 +34,12 @@ handle remainder => sub {
 
     my @numbers = split /[\s,;]+/, $input;
 
+    return unless @numbers > 1;
     return unless all { looks_like_number($_) } @numbers;
 
     my $count = 0;
     @numbers = map { 0 + $_ } grep { ++$count <= MAX_LIST_SIZE } @numbers;    # Normalize and limit list size.
-
+
```
Comment (GuiltyDolphin) [inline lib/DDG/Goodie/Sort.pm]: You've added some unneccesary whitespace here - you might want to set your editor to automatically strip trailing whitespace.

Comment (NaveenKarippai) [inline lib/DDG/Goodie/Sort.pm]: @GuiltyDolphin done! nvim user here :smile: The newline was inserted on purpose to beautify the code but following your suggestion, it has been removed.

Comment (GuiltyDolphin) [inline lib/DDG/Goodie/Sort.pm]: @NaveenKarippai The newline was already there - I meant you added some trailing whitespace, like `      `; please restore the newline (without the erroneous whitespace).

Comment (NaveenKarippai) [inline lib/DDG/Goodie/Sort.pm]: @GuiltyDolphin Please review the new changeset. If you were pointing to whitespace characters between (;) and (#) comment, those were there before I touched the file.

File: t/Sort.t [hunk 3]
```diff
@@ -32,6 +32,8 @@ ddg_goodie_test(
     'sort -3 -10 56 10' => build_test('-3, -10, 56, 10','ascendingly', '-10, -3, 10, 56'),
     'sort descending 10, -1, +5.3, -95, 1' => build_test('10, -1, 5.3, -95, 1', 'descendingly', '10, 5.3, 1, -1, -95'),
     'sort descending 10, -1, +5.3, -95, 1, 1e2' => build_test('10, -1, 5.3, -95, 1, 100', 'descendingly', '100, 10, 5.3, 1, -1, -95'),
+    'sort 1' => undef,
```
Comment (GuiltyDolphin) [inline t/Sort.t]: We only need one test here - these test exactly the same thing.

**Comment by moollaza** [comment]
@NaveenKarippai Thanks! This LGTM :+1:

File: t/Sort.t (hunk 3 above)
Comment (NaveenKarippai) [inline t/Sort.t]: @GuiltyDolphin Thanks. As this PR has already been merged, I shall remove this test case on PR https://github.com/duckduckgo/zeroclickinfo-goodies/pull/3157

**Pull request merged**